### DISCUSSION TITLE
how to add "move" to TWindow in TDialog (was: colored TButton)

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#126] [Q&A] [answered]
Hello,
I have put a seperate TEditor Window on top of a TDialog (because the TButton color problem).
Now, I can not "move" or "zoom" the TDialog.
My idea was, to put a "mover" on right bottom edge of the TDialog.
But I don't know, how to do so.

**Comment by magiblot** [comment, marked as answer]
This is because of the `growMode` and `flags` properties of `TWindow`/`TDialog`.

In `TWindow`, the default values are:

https://github.com/magiblot/tvision/blob/f369a73d835c010fc96f3fae505c73766157d13f/source/tvision/twindow.cpp#L43 https://github.com/magiblot/tvision/blob/f369a73d835c010fc96f3fae505c73766157d13f/source/tvision/twindow.cpp#L51

Conversely, the default values in `TDialog` are:

https://github.com/magiblot/tvision/blob/f369a73d835c010fc96f3fae505c73766157d13f/source/tvision/tdialog.cpp#L29-L30

Which is why you can no longer move or resize your editor window.

Therefore, the solution is to use the same values for `growMode` and `flags` as `TWindow`. You should set them in your class' constructor.

**Reply by paule32** [reply]
Thanks, this was/is help fully
